kotlin/display-kml/src/main/java/com/esri/arcgisruntime/sample/displaykml/MainActivity.kt

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2020 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.displaykml
+
+import android.os.Bundle
+import android.util.Log
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.layers.KmlLayer
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.loadable.Loadable
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.ogc.kml.KmlDataset
+import com.esri.arcgisruntime.portal.Portal
+import com.esri.arcgisruntime.portal.PortalItem
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG: String = MainActivity::class.java.getSimpleName()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        progressBar.visibility = View.VISIBLE
+
+        // create a map with the dark gray canvas basemap
+        val map = ArcGISMap(Basemap.Type.DARK_GRAY_CANVAS_VECTOR, 39.0, -98.0, 4);
+        // set the map to the map view
+        mapView.map = map
+        // prompt user to make a KML source selection when the app has loaded
+        map.addDoneLoadingListener {
+            if (map.loadStatus == LoadStatus.LOADED) {
+                Toast.makeText(applicationContext, R.string.user_prompt, Toast.LENGTH_LONG).show()
+                // hide progress bar once map has loaded
+                progressBar.visibility = View.GONE
+            }
+        }
+    }
+
+    /**
+     * Shows progress indicator if the KML Layer is loading, clears all operational layers and adds
+     * the KML layer to the map as an operational layer.
+     *
+     * @param kmlLayer to add to the map
+     */
+    fun display(kmlLayer: KmlLayer) {
+        // show progress indicator if kml dataset is loading
+        progressBar.visibility = View.VISIBLE
+        kmlLayer.addDoneLoadingListener {
+            if (kmlLayer.loadStatus == LoadStatus.LOADED) {
```

Comment:
we probably want to hide the progress bar even if its FAILED_TO_LOAD, because we don't want to imply its still working on something if it failed

Reply:
I've covered this by hiding the progress indicator in the reportErrors function